File: Data Structures & Algorithms/anagram-groups/submission-3.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res = {}

        for string in strs:
            key = self.generateKey(string)

            if key not in res:
                res[key] = []
            res[key].append(string)

        return list(res.values())

    def generateKey(self, string: str):
        count = [0] * 26
        for char in string:
            count[ord(char) - ord('a')] += 1
        
        return tuple(count)

File: Data Structures & Algorithms/concatenation-of-array/submission-3.cs
public class Solution {
    public int[] GetConcatenation(int[] nums) {
        int[] res = new int[nums.Length * 2];

        int idx = 0;

        while(idx < nums.Length) {
            res[idx] = nums[idx];
            res[idx + nums.Length] = nums[idx];
            idx++;
        }


        return res;

    }
} 